game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/FireRoundStepsBuilderTest.java

```diff
@@ -0,0 +1,147 @@
+                        (arg1) ->
+                            List.of(
+                                FiringGroup.groupBySuicideOnHit(
+                                        "army", List.of(givenAnyUnit()), List.of(mock(Unit.class)))
+                                    .get(0),
+                                FiringGroup.groupBySuicideOnHit(
+                                        "spies", List.of(givenAnyUnit()), List.of(mock(Unit.class)))
+                                    .get(0)))
+                    .selectCasualties((arg1, arg2) -> new CasualtyDetails())
+                    .side(OFFENSE)
+                    .battleState(
+                        FakeBattleState.givenBattleStateBuilder()
+                            .attacker(attacker)
+                            .defender(defender)
+                            .build())
+                    .build()));
+
+    final List<String> expected =
+        List.of(
+            "attacker army" + FIRE_SUFFIX,
+            "defender" + SELECT_PREFIX + "army" + CASUALTIES_SUFFIX,
+            "defender" + REMOVE_PREFIX + "army" + CASUALTIES_SUFFIX,
+            "attacker spies" + FIRE_SUFFIX,
+            "defender" + SELECT_PREFIX + "spies" + CASUALTIES_SUFFIX,
+            "defender" + REMOVE_PREFIX + "spies" + CASUALTIES_SUFFIX);
+    assertThat(names, is(expected));
+  }
+
+  @Test
+  @DisplayName("With the default name UNITS, the step names should be readable")
+  void getNamesWithDefaultUnitName() {
+    final GamePlayer attacker = mock(GamePlayer.class);
+    when(attacker.getName()).thenReturn("attacker");
+    final GamePlayer defender = mock(GamePlayer.class);
+    when(defender.getName()).thenReturn("defender");
+    final List<String> names =
+        getStepNames(
+            FireRoundStepsBuilder.buildSteps(
+                FireRoundStepsBuilder.Parameters.builder()
+                    .battleActions(mock(BattleActions.class))
+                    .returnFire(MustFightBattle.ReturnFire.ALL)
+                    .roll((arg1, arg2) -> new DiceRoll())
+                    .firingGroupSplitter(
+                        (arg1) ->
+                            FiringGroup.groupBySuicideOnHit(
+                                UNITS, List.of(givenAnyUnit()), List.of(mock(Unit.class))))
+                    .selectCasualties((arg1, arg2) -> new CasualtyDetails())
+                    .side(OFFENSE)
+                    .battleState(
+                        FakeBattleState.givenBattleStateBuilder()
+                            .attacker(attacker)
+                            .defender(defender)
+                            .build())
+                    .build()));
+
+    final List<String> expected =
+        List.of(
+            "attacker" + FIRE_SUFFIX,
+            "defender" + SELECT_PREFIX + CASUALTIES_WITHOUT_SPACE_SUFFIX,
+            "defender" + REMOVE_PREFIX + CASUALTIES_WITHOUT_SPACE_SUFFIX);
+    assertThat(names, is(expected));
```

Comment:
Thoughts to use the iterable matcher here? You could avoid created the intermediate list and hopefully the failure message would be more user-friendly and tell you exactly how the lists mismatched, EG:

```
assertThat(names, IsIterableContainingInOrder.contains(
    "attacker" + FIRE_SUFFIX,
            "defender" + SELECT_PREFIX + CASUALTIES_WITHOUT_SPACE_SUFFIX,
            "defender" + REMOVE_PREFIX + CASUALTIES_WITHOUT_SPACE_SUFFIX));

```

Reply:
I tried the `IsIterableContainingInOrder` and I think the failure message from `is` is actually better.  With `is`, it shows the entire `expected` and `actual` array.  But with `IsIterableContainingInOrder`, it only shows the complete `expected` array and the first failure item in the `actual` array.  I think seeing both complete arrays is useful.